src/lib/protocols/soap.c

```diff
@@ -37,6 +44,28 @@ void ndpi_search_soap(struct ndpi_detection_module_struct *ndpi_struct,
 
   NDPI_LOG_DBG(ndpi_struct, "search soap\n");
 
+  if (packet->parsed_lines == 0)
+  {
+    ndpi_parse_packet_line_info(ndpi_struct, flow);
+  }
+
+  if (packet->parsed_lines > 0)
+  {
+    size_t i;
+
+    for (i = 0; i < packet->parsed_lines && packet->line[i].len > 0; ++i)
+    {
+      if (LINE_STARTS(packet->line[i], "SOAPAction") != 0)
+      {
+        ndpi_int_soap_add_connection(ndpi_struct, flow);
+        return;
+      }
+    }
+
+    NDPI_EXCLUDE_PROTO(ndpi_struct, flow);
+    return;
+  }
+
```

Comment:
Shouldn't **all** this new code be within an `if (flow->detected_protocol_stack[0] == NDPI_PROTOCOL_HTTP)` kind of check?

Reply:
True, but the http dissector has an issue detecting some POSTs. I can try to fix it here or in a follow-up.

Reply:
As you prefer... I noticed that because the attached flow is no more classified as SOAP
[soap3.zip](https://github.com/ntop/nDPI/files/8933722/soap3.zip)

Reply:
I forgot that the packet line parse just searches for `\r\n` ...
`if (flow->detected_protocol_stack[0] == NDPI_PROTOCOL_HTTP)` would be the better solution, but needs some http dissector fixes e.g. handling HTTP POSTs like the one `tests/pcap/soap.pcap`.